### PR TITLE
Use UPower interface to detect lid

### PR DESCRIPTION
**Pull request by crmlt**
- use UPower to check if lid switch is present

### CODE REVIEW

**Comment by augustin-b** [comment]
@donadigo Another concept requested here https://github.com/elementary/switchboard-plug-power/pull/60

What shall I do?

**Comment by donadigo** [comment]
@augustin-b I reached out to kay so that we can agree or disagree on this. Just holding this branch off right now.

**Comment by augustin-b** [comment]
@donadigo Np, here is branch based on your requested changes.. just to compare...

https://github.com/augustin-b/switchboard-plug-power/commit/e012419ecedb6796c86415131625e68f3624cfe3

File: src/Lid.vala [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        construct {
+            try {
+                upower = Bus.get_proxy_sync (BusType.SYSTEM, DBUS_UPOWER_NAME, DBUS_UPOWER_PATH, DBusProxyFlags.NONE);
+                debug ("Connection to UPower bus established");
+            } catch (Error e) {
+                critical ("Connecting to UPower bus failed: %s", e.message);
+            }
+        }
+
+        public bool is_present () {
```
Comment (augustin-b) [inline src/Lid.vala]: Should I use try and catch error here?

**Comment by kvdz** [comment]
i think it looks good. only opimalisation would be to connect once to the upower root path over battery lid and power supply classes instead of 3 times. i disagree with donadigo about making a class for the lid. i think it must be a class. otherwise there is a bad seperation of concerns.

**Comment by augustin-b** [comment]
Can this be merged now so we could optimize DBUS connection later?

**Comment by danrabbit** [comment]
I'm going to close this PR since it looks like work on this was stalled and it now conflicts with master. If work here resumes, please feel free to reopen :)

**Comment by crmlt** [comment]
@danrabbit Np

**Pull request closed without merging**